src/lib.rs

```diff
@@ -440,11 +440,19 @@ impl KernelResolver {
         kernel_image: &str,
         cache_holder: &CacheHolder,
     ) -> Result<KernelResolver, Error> {
-        let ksymresolver = cache_holder.get_ksym_cache().get_resolver(kallsyms)?;
-        let kernelresolver = ElfResolver::new(kernel_image, 0, cache_holder)?;
+        let ksymresolver = cache_holder.get_ksym_cache().get_resolver(kallsyms);
+        let kernelresolver = ElfResolver::new(kernel_image, 0, cache_holder);
```

Comment:
naming nit. It's not clear what's the difference between ksymresolver and kernelresolver, tbh. maybe at least call kernelresolve an kelfresolver or something along those lines to point out that it's relying on ELF file?